staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go

```diff
@@ -156,13 +156,20 @@ func (ss *scaleSet) newVMSSVirtualMachinesCache() (*timedCache, error) {
 					}
 
 					computerName := strings.ToLower(*vm.OsProfile.ComputerName)
-					localCache.Store(computerName, &vmssVirtualMachinesEntry{
+					vmssVMCacheEntry := &vmssVirtualMachinesEntry{
 						resourceGroup:  resourceGroup,
 						vmssName:       ssName,
 						instanceID:     to.String(vm.InstanceID),
 						virtualMachine: &vm,
 						lastUpdate:     time.Now().UTC(),
-					})
+					}
+					// set cache entry to nil when the VM is under deleting.
+					if vm.VirtualMachineScaleSetVMProperties != nil &&
+						strings.EqualFold(to.String(vm.VirtualMachineScaleSetVMProperties.ProvisioningState), vmProvisioningStateDeleting) {
```

Comment:
could you reference the value as `compute.ProvisioningStateDeleting`?

Reply:
Good catch. Fixed now.